### DISCUSSION TITLE
C# Markup: How to access 'model' from an event?

### DISCUSSION THREAD

**Discussion by kucint** [unoplatform/uno#15124] [Q&A] [answered]
I want to perform some action on "LostFocus".
The action is defined in the "model" class.

I achieve the goal by subscribing an event.
**Question 1**: Is it a right way to go?  Or I should rather do it differently? How? Maybe using Behaviors?

**Question 2**: Following code crashes: `vm` is `null `when call  `vm.DoSomething()` is executed.
I assume that some "binding" is missing. 
How should I do it?

```
public MainPage() => this
            .DataContext(new BindableMainModel(), (page, vm) => page
            .Content(
                new TextBox()
                .Assign((TextBox c) =>
                {
                    c.LostFocus += (s, e) =>
                    {
                        vm.DoSomething();
                    };
                }));

internal partial record Model
{
    public async ValueTask DoSomething()    {  }
}
```

**Comment by kucint** [comment, marked as answer]
I figured it out:
```
  new TextBox()
  .Assign((TextBox c) =>
  {
      c.LostFocus += (s, e) =>
      {
          object? parameter = null;
          if (DataContext is BindableModel m)
          {
              if (m.ValidateParameters.CanExecute(parameter))
                  m.ValidateParameters.Execute(parameter);
          }
      };
  }));
```
but the **Question 1** still remains open: Is it a right approach?

**Reply by dansiegel** [reply]
So to help answer your questions... yes the way that you have it currently is the easiest way to add support for hooking into an event and executing a Command. Better Event Support is something we would like to add and will most likely use Behaviors behind the scenes to ensure that Event Handlers are also removed as the Control is destroyed.

**Reply by kucint** [reply]
Thanks @dansiegel,
Does it mean that this new way of handling events is not available for now and is still under the development?
if so, what is approx. estimated time of delivery?
Does an issue/task associated to this topic already exist? I would like to follow it!

BTW. I do really love C# markup. appreciate your job!

**Reply by francoistanguay** [reply]
Assigning a variable and registering to the event like you're doing is currently the recommended approach.

Even though we're exploring other options for the future, I believe they do have drawbacks in terms of ease of use and discoverability so they aren't planned at this time.

If that changes, we'll make sure to let the community know by updating the various related issues and discussions.

**Reply by kucint** [reply]
thanks for info.
